codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessage.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+/**
+ * RedisMessage is base interface for codec-redis.
+ */
+public interface RedisMessage {
+
+    /**
+     * Returns the {@link RedisMessageType}.
+     *
+     * @return the message type.
+     */
+    RedisMessageType type();
```

Comment:
The main use case for this appears to be in the encoder. However at this point you already have encode methods broken down by the type hierarchy (in `writeRedisMessage`) where it seems like this information should be known. In order to add support for new types the encoder must also be modified and so I'm not sure if this method provides much value on this interface. Can you clarify any merits of this method that I am missing?